unittests/backends/QPPTester.cpp

```diff
@@ -26,6 +26,20 @@ void print_state(const qpp::ket &stateVector) {
   std::cout << "]\n";
 }
 
+void print_state(const qpp::cmat &densityMatrix) {
+  std::cout << "state = [";
+  auto rows = densityMatrix.rows();
+  auto cols = densityMatrix.cols();
+  for (auto rdx = 0; rdx < rows; rdx++) {
```

Comment:
Looks old school. Does this not support standard iterators?

Reply:
There may be a cleaner way to do this with Eigen::IOFormat , I just wrote this up quickly as a way to format the output as a nice, square matrix